src/lib/support/PersistedCounter.h

```diff
@@ -206,12 +206,29 @@ class PersistedCounter : public MonotonicallyIncreasingCounter<T>
         aStartValue = Encoding::LittleEndian::HostSwap<T>(valueLE);
 
 #if CHIP_CONFIG_PERSISTED_COUNTER_DEBUG_LOGGING
-        ChipLogDetail(EventLogging, "PersistedCounter::ReadStartValue() aStartValue 0x%x", aStartValue);
+        // Compiler should optimize these branches.
```

Comment:
Why not just `if constexpr` instead of "should" comments?  We're c++17 now.

Reply:
updated all the other spots in PersistedCounter.h